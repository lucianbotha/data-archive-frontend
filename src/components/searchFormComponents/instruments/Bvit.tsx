import * as React from "react";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";
import { IBVIT } from "../../../utils/ObservationQueryParameters";

interface BvitProps {
  bvit: IBVIT;
  onChange: (value: any) => void;
}

/**
 * A form for selecting HRS-related search parameters.
 */
const Bvit = (props: BvitProps) => {
  const { onChange, bvit } = props;

  // Function for handling change events
  const change = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const name = e.currentTarget.name;
    const value = e.currentTarget.value;
    onChange({
      ...bvit,
      [name]: value
    });
  };

  const modes = ["Imaging", "Streaming"];
  const filters = ["B", "H-alpha", "Open", "R", "U", "V"];

  return (
    <MainGrid>
      <SubGrid>
        <p>Mode</p>
        <SelectField name={"mode"} onChange={change}>
          <AnyOption />
          {modes.map(mode => (
            <option key={mode} value={mode}>
              {mode}
            </option>
          ))}
        </SelectField>
      </SubGrid>
      <SubGrid>
        <p>Filter</p>
        <SelectField name={"filter"} onChange={change}>
          <AnyOption />
          {filters.map(filter => (
            <option key={filter} value={filter}>
              {filter}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};

export default Bvit;
