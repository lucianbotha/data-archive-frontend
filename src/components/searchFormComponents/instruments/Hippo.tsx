import * as React from "react";
import { IHIPPO } from "../../../utils/ObservationQueryParameters";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import InputField from "../../basicComponents/InputField";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";

interface IHippoProps {
  hippo: IHIPPO;
  onChange: (value: any) => void;
}

/**
 * A form for selecting Hippo-related search parameters.
 */
const Hippo = (props: IHippoProps) => {
  const { onChange, hippo } = props;
  const change = (
    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    const name = e.target.name;
    const value = e.target.value;
    onChange({
      ...hippo,
      [name]: value
    });
  };
  const filters = ["B", "H-alpha", "Open", "R", "U", "V"];
  return (
    <MainGrid>
      <SubGrid>
        <p>Exposure time</p>
        <InputField name={"exposureTime"} onChange={change} />
      </SubGrid>
      <SubGrid>
        <p>Filter</p>
        <SelectField name={"filter"} onChange={change}>
          <AnyOption />
          {filters.map(filter => (
            <option key={filter} value={filter}>
              {filter}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};
export default Hippo;
