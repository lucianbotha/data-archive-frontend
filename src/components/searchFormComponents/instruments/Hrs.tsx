import * as React from "react";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";
import { IHRS } from "../../../utils/ObservationQueryParameters";

interface IHrsProps {
  hrs: IHRS;
  onChange: (value: any) => void;
}

/**
 * A form for selecting HRS-related search parameters.
 * @param props
 * @constructor
 */
const Hrs = (props: IHrsProps) => {
  const { onChange, hrs } = props;

  // Function for handling change events
  const change = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const name = e.currentTarget.name;
    const value = e.currentTarget.value;
    onChange({
      ...hrs,
      [name]: value
    });
  };

  const modes = [
    "High Resolution",
    "High Stability",
    "Int Cal Fibre",
    "Low Resolution",
    "Medium Resolution"
  ];

  return (
    <MainGrid>
      <SubGrid>
        <p>Mode</p>
        <SelectField name={"mode"} onChange={change}>
          <AnyOption />
          {modes.map(mode => (
            <option key={mode} value={mode.toUpperCase()}>
              {mode}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};
export default Hrs;
