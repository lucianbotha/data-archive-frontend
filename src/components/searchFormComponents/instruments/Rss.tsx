import * as React from "react";
import { IRSS } from "../../../utils/ObservationQueryParameters";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";

interface IRssProps {
  rss: IRSS;
  onChange: (value: any) => void;
}

/**
 * A form selecting RSS-related search parameters.
 */
const Rss = (props: IRssProps) => {
  const { onChange, rss } = props;

  // Function for handling change events
  const change = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const name = e.currentTarget.name;
    const value = e.currentTarget.value;
    onChange({
      ...rss,
      [name]: value
    });
  };

  const modes = [
    "Fabry Perot",
    "FP polarimetry",
    "Imaging",
    "Polarimetric imaging",
    "MOS",
    "MOS polarimetry",
    "Spectropolarimetry",
    "Spectroscopy"
  ];

  const detectorModes = [
    "Normal",
    "Frame Transfer",
    "Slot Mode",
    "Shuffle",
    "Drift Scan"
  ];

  return (
    <MainGrid>
      <SubGrid>
        <p>Mode</p>
        <SelectField name={"mode"} onChange={change}>
          <AnyOption />
          {modes.map(mode => (
            <option key={mode} value={mode}>
              {mode}
            </option>
          ))}
        </SelectField>
      </SubGrid>
      <SubGrid>
        <p>Detector Mode</p>
        <SelectField name={"detectorMode"} onChange={change}>
          <AnyOption />
          {detectorModes.map(detectorMode => (
            <option key={detectorMode} value={detectorMode}>
              {detectorMode}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};
export default Rss;
