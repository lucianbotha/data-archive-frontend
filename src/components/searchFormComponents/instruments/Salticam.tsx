import * as React from "react";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";
import { ISalticam } from "../../../utils/ObservationQueryParameters";

interface ISalticamProps {
  salticam: ISalticam;
  onChange: (value: any) => void;
}

/**
 * A form for selecting Salticam-related search parameters.
 */
const Salticam = (props: ISalticamProps) => {
  const { onChange, salticam } = props;

  const change = (e: React.FormEvent<HTMLSelectElement>) => {
    const name = e.currentTarget.name;
    const value = e.currentTarget.value;
    onChange({
      ...salticam,
      [name]: value
    });
  };

  const detectorModes = ["Normal", "Slot Mode", "Drift Scan", "Frame Transfer"];

  return (
    <MainGrid>
      <SubGrid>
        <p>Detector Mode</p>
        <SelectField name={"detectorMode"} onChange={change}>
          <AnyOption />
          {detectorModes.map(detectorMode => (
            <option key={detectorMode} value={detectorMode}>
              {detectorMode}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};

export default Salticam;
