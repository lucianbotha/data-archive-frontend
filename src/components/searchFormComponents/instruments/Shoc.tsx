import * as React from "react";
import { ISHOC } from "../../../utils/ObservationQueryParameters";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import InputField from "../../basicComponents/InputField";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";

interface IShocProps {
  shoc: ISHOC;
  onChange: (value: any) => void;
}

/**
 * A form for selecting Shoc-related search parameters.
 */
const Shoc = (props: IShocProps) => {
  const { onChange, shoc } = props;
  const change = (
    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    const name = e.target.name;
    const value = e.target.value;
    onChange({
      ...shoc,
      [name]: value
    });
  };
  const filters = ["B", "H-alpha", "Open", "R", "U", "V"];
  return (
    <MainGrid>
      <SubGrid>
        <p>Exposure time</p>
        <InputField name={"exposureTime"} onChange={change} />
      </SubGrid>
      <SubGrid>
        <p>Filter</p>
        <SelectField name={"filter"} onChange={change}>
          <AnyOption />
          {filters.map(filter => (
            <option key={filter} value={filter}>
              {filter}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};
export default Shoc;
