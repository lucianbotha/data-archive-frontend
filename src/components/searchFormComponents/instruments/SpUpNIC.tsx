import * as React from "react";
import { ISpUpNIC } from "../../../utils/ObservationQueryParameters";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import InputField from "../../basicComponents/InputField";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";

interface ISpUpNICProps {
  spUpNic: ISpUpNIC;
  onChange: (value: any) => void;
}

/**
 * A form for selecting SpUpNIC-related search parameters.
 */
const SpUpNIC = (props: ISpUpNICProps) => {
  const { onChange, spUpNic } = props;
  const change = (
    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    const name = e.target.name;
    const value = e.target.value;
    onChange({
      ...spUpNic,
      [name]: value
    });
  };
  const filters = ["B", "H-alpha", "Open", "R", "U", "V"];
  return (
    <MainGrid>
      <SubGrid>
        <p>Exposure time</p>
        <InputField name={"exposureTime"} onChange={change} />
      </SubGrid>
      <SubGrid>
        <p>Filter</p>
        <SelectField name={"filter"} onChange={change}>
          <AnyOption />
          {filters.map(filter => (
            <option key={filter} value={filter}>
              {filter}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};
export default SpUpNIC;
