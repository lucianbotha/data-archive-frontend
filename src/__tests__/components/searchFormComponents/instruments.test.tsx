import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import Bvit from "../../../components/searchFormComponents/instruments/Bvit";
import Hippo from "../../../components/searchFormComponents/instruments/Hippo";
import Hrs from "../../../components/searchFormComponents/instruments/Hrs";
import Rss from "../../../components/searchFormComponents/instruments/Rss";
import Salticam from "../../../components/searchFormComponents/instruments/Salticam";
import Shoc from "../../../components/searchFormComponents/instruments/Shoc";
import SpUpNIC from "../../../components/searchFormComponents/instruments/SpUpNIC";

const onChange = jest.fn();
describe("Grids ", () => {
  it("should render all grids", () => {
    expect(
      shallow(<Bvit bvit={{ errors: {}, name: "BVIT" }} onChange={onChange} />)
    ).toBeDefined();
    expect(
      shallow(
        <Hippo hippo={{ errors: {}, name: "HIPPO" }} onChange={onChange} />
      )
    ).toBeDefined();
    expect(
      shallow(<Hrs hrs={{ errors: {}, name: "HRS" }} onChange={onChange} />)
    ).toBeDefined();
    expect(
      shallow(<Rss rss={{ errors: {}, name: "RSS" }} onChange={onChange} />)
    ).toBeDefined();
    expect(
      shallow(
        <Salticam
          salticam={{ errors: {}, name: "Salticam" }}
          onChange={onChange}
        />
      )
    ).toBeDefined();
    expect(
      shallow(<Shoc shoc={{ errors: {}, name: "SHOC" }} onChange={onChange} />)
    ).toBeDefined();
    expect(
      shallow(
        <SpUpNIC
          spUpNic={{ errors: {}, name: "SpUpNIC" }}
          onChange={onChange}
        />
      )
    ).toBeDefined();
  });

  it("should render all grids correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(
        shallow(
          <Bvit bvit={{ errors: {}, name: "BVIT" }} onChange={onChange} />
        )
      )
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(
          <Hippo hippo={{ errors: {}, name: "HIPPO" }} onChange={onChange} />
        )
      )
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(<Hrs hrs={{ errors: {}, name: "HRS" }} onChange={onChange} />)
      )
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(<Rss rss={{ errors: {}, name: "RSS" }} onChange={onChange} />)
      )
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(
          <Salticam
            salticam={{ errors: {}, name: "Salticam" }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(
          <Shoc shoc={{ errors: {}, name: "SHOC" }} onChange={onChange} />
        )
      )
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(
          <SpUpNIC
            spUpNic={{ errors: {}, name: "SpUpNIC" }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
