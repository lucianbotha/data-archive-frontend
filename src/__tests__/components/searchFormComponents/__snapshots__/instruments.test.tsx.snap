// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grids  should render all grids correctly 1`] = `
<styled.div>
  <styled.div>
    <p>
      Mode
    </p>
    <SelectField
      name="mode"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="Imaging"
        value="Imaging"
      >
        Imaging
      </option>
      <option
        key="Streaming"
        value="Streaming"
      >
        Streaming
      </option>
    </SelectField>
  </styled.div>
  <styled.div>
    <p>
      Filter
    </p>
    <SelectField
      name="filter"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="B"
        value="B"
      >
        B
      </option>
      <option
        key="H-alpha"
        value="H-alpha"
      >
        H-alpha
      </option>
      <option
        key="Open"
        value="Open"
      >
        Open
      </option>
      <option
        key="R"
        value="R"
      >
        R
      </option>
      <option
        key="U"
        value="U"
      >
        U
      </option>
      <option
        key="V"
        value="V"
      >
        V
      </option>
    </SelectField>
  </styled.div>
</styled.div>
`;

exports[`Grids  should render all grids correctly 2`] = `
<styled.div>
  <styled.div>
    <p>
      Exposure time
    </p>
    <InputField
      name="exposureTime"
      onChange={[Function]}
    />
  </styled.div>
  <styled.div>
    <p>
      Filter
    </p>
    <SelectField
      name="filter"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="B"
        value="B"
      >
        B
      </option>
      <option
        key="H-alpha"
        value="H-alpha"
      >
        H-alpha
      </option>
      <option
        key="Open"
        value="Open"
      >
        Open
      </option>
      <option
        key="R"
        value="R"
      >
        R
      </option>
      <option
        key="U"
        value="U"
      >
        U
      </option>
      <option
        key="V"
        value="V"
      >
        V
      </option>
    </SelectField>
  </styled.div>
</styled.div>
`;

exports[`Grids  should render all grids correctly 3`] = `
<styled.div>
  <styled.div>
    <p>
      Mode
    </p>
    <SelectField
      name="mode"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="High Resolution"
        value="HIGH RESOLUTION"
      >
        High Resolution
      </option>
      <option
        key="High Stability"
        value="HIGH STABILITY"
      >
        High Stability
      </option>
      <option
        key="Int Cal Fibre"
        value="INT CAL FIBRE"
      >
        Int Cal Fibre
      </option>
      <option
        key="Low Resolution"
        value="LOW RESOLUTION"
      >
        Low Resolution
      </option>
      <option
        key="Medium Resolution"
        value="MEDIUM RESOLUTION"
      >
        Medium Resolution
      </option>
    </SelectField>
  </styled.div>
</styled.div>
`;

exports[`Grids  should render all grids correctly 4`] = `
<styled.div>
  <styled.div>
    <p>
      Mode
    </p>
    <SelectField
      name="mode"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="Fabry Perot"
        value="Fabry Perot"
      >
        Fabry Perot
      </option>
      <option
        key="FP polarimetry"
        value="FP polarimetry"
      >
        FP polarimetry
      </option>
      <option
        key="Imaging"
        value="Imaging"
      >
        Imaging
      </option>
      <option
        key="Polarimetric imaging"
        value="Polarimetric imaging"
      >
        Polarimetric imaging
      </option>
      <option
        key="MOS"
        value="MOS"
      >
        MOS
      </option>
      <option
        key="MOS polarimetry"
        value="MOS polarimetry"
      >
        MOS polarimetry
      </option>
      <option
        key="Spectropolarimetry"
        value="Spectropolarimetry"
      >
        Spectropolarimetry
      </option>
      <option
        key="Spectroscopy"
        value="Spectroscopy"
      >
        Spectroscopy
      </option>
    </SelectField>
  </styled.div>
  <styled.div>
    <p>
      Detector Mode
    </p>
    <SelectField
      name="detectorMode"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="Normal"
        value="Normal"
      >
        Normal
      </option>
      <option
        key="Frame Transfer"
        value="Frame Transfer"
      >
        Frame Transfer
      </option>
      <option
        key="Slot Mode"
        value="Slot Mode"
      >
        Slot Mode
      </option>
      <option
        key="Shuffle"
        value="Shuffle"
      >
        Shuffle
      </option>
      <option
        key="Drift Scan"
        value="Drift Scan"
      >
        Drift Scan
      </option>
    </SelectField>
  </styled.div>
</styled.div>
`;

exports[`Grids  should render all grids correctly 5`] = `
<styled.div>
  <styled.div>
    <p>
      Detector Mode
    </p>
    <SelectField
      name="detectorMode"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="Normal"
        value="Normal"
      >
        Normal
      </option>
      <option
        key="Slot Mode"
        value="Slot Mode"
      >
        Slot Mode
      </option>
      <option
        key="Drift Scan"
        value="Drift Scan"
      >
        Drift Scan
      </option>
      <option
        key="Frame Transfer"
        value="Frame Transfer"
      >
        Frame Transfer
      </option>
    </SelectField>
  </styled.div>
</styled.div>
`;

exports[`Grids  should render all grids correctly 6`] = `
<styled.div>
  <styled.div>
    <p>
      Exposure time
    </p>
    <InputField
      name="exposureTime"
      onChange={[Function]}
    />
  </styled.div>
  <styled.div>
    <p>
      Filter
    </p>
    <SelectField
      name="filter"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="B"
        value="B"
      >
        B
      </option>
      <option
        key="H-alpha"
        value="H-alpha"
      >
        H-alpha
      </option>
      <option
        key="Open"
        value="Open"
      >
        Open
      </option>
      <option
        key="R"
        value="R"
      >
        R
      </option>
      <option
        key="U"
        value="U"
      >
        U
      </option>
      <option
        key="V"
        value="V"
      >
        V
      </option>
    </SelectField>
  </styled.div>
</styled.div>
`;

exports[`Grids  should render all grids correctly 7`] = `
<styled.div>
  <styled.div>
    <p>
      Exposure time
    </p>
    <InputField
      name="exposureTime"
      onChange={[Function]}
    />
  </styled.div>
  <styled.div>
    <p>
      Filter
    </p>
    <SelectField
      name="filter"
      onChange={[Function]}
    >
      <AnyOption />
      <option
        key="B"
        value="B"
      >
        B
      </option>
      <option
        key="H-alpha"
        value="H-alpha"
      >
        H-alpha
      </option>
      <option
        key="Open"
        value="Open"
      >
        Open
      </option>
      <option
        key="R"
        value="R"
      >
        R
      </option>
      <option
        key="U"
        value="U"
      >
        U
      </option>
      <option
        key="V"
        value="V"
      >
        V
      </option>
    </SelectField>
  </styled.div>
</styled.div>
`;
